src/nettests/index.js

```diff
@@ -1,9 +1,106 @@
+import camelCase from 'camelcase'
+
+import percentage from '../cli/output/percentage'
+import wait from '../cli/output/wait'
+
+import {
+  Result,
+  Measurement
+} from '../config/db'
+
 import HTTPHeaderFieldManipulation from './http-header-field-manipulation'
 import HTTPInvalidRequestLine from './http-invalid-request-line'
 import WebConnectivity from './web-connectivity'
 import NDT from './ndt'
 import Base from './base'
 
+export const makeOoni = () => {
+  let dbOperations = [],
+      progress = null,
+      reportId = null,
+      uploaded = false,
+      localReportId = null,
+      isMk = false
+
+  let measurement = Measurement.build({
+    state: 'active'
+  })
+
+  const init = (nt) => {
+    // XXX we use duck typing to see if the argument is a measurement Kit
+    // nettest. Maybe we should change this.
+    if (nt.test && nt.name && nt.setOptions) isMk = true
```

Comment:
What is exactly duck typing? A way of doing run-time-type information?

Reply:
(Wouldn't it be possible to explicitly add a field for this purpose?)

Reply:
Duck typing == if it quacks like a duck it much be a duck. I think actually it's better to use duck typing here as it means that as long as people expose those attributes (i.e. implement a MK compatible API) we should be good to go.

Reply:
LOL, alright, I don't care particularly to argue for one solution or another